lib/server/services/eosReport/EosReportService.js

```diff
@@ -0,0 +1,107 @@
+/**
+ *  @license
+ *  Copyright CERN and copyright holders of ALICE O2. This software is
+ *  distributed under the terms of the GNU General Public License v3 (GPL
+ *  Version 3), copied verbatim in the file "COPYING".
+ *
+ *  See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ *  In applying this license CERN does not waive the privileges and immunities
+ *  granted to it by virtue of its status as an Intergovernmental Organization
+ *  or submit itself to any jurisdiction.
+ */
+const { formatEosReport } = require('./formatEosReport.js');
+const { getUserOrFail } = require('../user/getUserOrFail.js');
+const { getTagsByText } = require('../tag/getTagsByText.js');
+const { getShiftFromTimestamp } = require('./getShiftFromTimestamp.js');
+const { getLogsByTagsInPeriod } = require('../log/getLogsByTagsInPeriod.js');
+const { logService } = require('../log/LogService.js');
+
+/**
+ * @typedef Shift
+ * @property {number} start the start of the shift (UNIX timestamp in ms)
+ * @property {number} end the end of the shift (UNIX timestamp in ms)
+ * @property {'Morning'|'Afternoon'|'Night'} period the period of the shift
+ */
+
+/**
+ * @typedef EosReport
+ * @property {string} type
+ * @property {number} shiftStart start of the shift (UNIX timestamp in ms)
+ * @property {User} shifter
+ * @property {Log[]} issuesLogEntries
+ * @property {string} [traineeName]
+ * @property {string} [issues]
+ * @property {string} [shiftFlow]
+ * @property {string} [infoFromPreviousShifter
+ * @property {string} [infoForNextShifter]
+ * @property {string} [infoForRmRc]
+ */
+
+/**
+ * Service to generate end of shift report
+ */
+class EosReportService {
+    /**
+     * Constructor
+     */
+    constructor() {
+        this._issuesLogEntriesTags = ['DCS', 'ECS', 'QC/PDP', 'SL', 'SLIMOS'];
+    }
+
+    /**
+     * Create an end of shift report log
+     *
+     * @param {string} reportType the type of EOS report
+     * @param {EosReportCreationRequest} reportCreationRequest the data of the EOS report
+     * @param {UserIdentifier} shifterUserIdentifier the shifter creatiing the EOS report
+     * @return {Promise<number>} resolves with the created log entry
+     */
+    async createLogEntry(reportType, reportCreationRequest, shifterUserIdentifier) {
+        const shifter = await getUserOrFail(shifterUserIdentifier);
+        const shift = getShiftFromTimestamp(reportCreationRequest.shiftStart);
```

Comment:
While testing, a part of the title was saved as "undefined": `End of shift report - ECS - Invalid Date undefined`.
Seems to be an issue with the timestamp. For my test I used:
`"shiftStart": 1679302656000`